### PR TITLE
CI failures

### PR DESCRIPTION
**Pull request by anissa111**
Addresses #207 

### Summary of changes directly addressing #207:
- To address the broken OPeNDAP url: adds a try/catch before attempting remainder of test (in addition to @ocefpaf's contribution in https://github.com/UXARRAY/uxarray/pull/209)
- Does not touch reflected list deprecation in numba, that will be a separate PR since it's not currently causing tests to fail
- To address failing upstream ci caused by pip installs overriding conda pins:
    - removes upstream pip installs from `upstream-dev-environment.yml`
    - take dependency install out of `conda_setup` step
    - add `Install select upstream dependencies` step
    - re-add conda dependencies install _after_ pip dependencies

### Bonus changes:
- adds a `conda list` step to `upstream-dev-ci.yml` and `ci.yml`
- updated upstream CI to run on python 3.10
- added python 3.10 to version matrix for regular CI, too
- renames environment in `upstream-dev-ci.yml` to be `uxarray_upstream_build` instead of `uxarray_build` to be distinct from the environment name in `environment.yml` and `ci.yml`
- changes pip install from git to zip for speed
- uses conda instead of mamba for more consistent environment solving only for upstream ci (errors persist if mamba is used)

See passing upstream ci for this branch here: https://github.com/UXARRAY/uxarray/actions/runs/4021274777/jobs/6910001710

### CODE REVIEW

File: ci/upstream-dev-environment.yml [hunk 1]
```diff
@@ -1,17 +1,12 @@
-name: uxarray_build
+name: uxarray_upstream_build
 channels:
   - conda-forge
 dependencies:
   - netcdf4
   - numba
-  # pinning the numpy version for numba compatibility
-  - numpy
+  - numpy>=1.2, <1.25
```
Comment (erogluorhan) [inline ci/upstream-dev-environment.yml:7]: Is this version pinning needed? I think it was resolved a few releases ago for the regular CI.

**Review by erogluorhan — COMMENTED** [review]
It looks good overall but a single inline comment

File: ci/upstream-dev-environment.yml (hunk 1 above)
Comment (anissa111) [reply ci/upstream-dev-environment.yml:7]: Oh good point! It was something I was trying along the way. Let me revert this and see if everything still holds

**Review by erogluorhan — APPROVED** [review]

**Review by philipc2 — APPROVED** [review]
Looks good!

**Pull request merged**